llvm/test/CodeGen/PowerPC/builtins-ppc-xlcompat-LoadReserve-StoreCond-64bit-only.ll

```diff
@@ -26,7 +26,7 @@ define dso_local i64 @test_stdcx(ptr %a, i64 %b) {
 ; CHECK-NEXT:    stdcx. 4, 0, 3
 ; CHECK-NEXT:    mfocrf 3, 128
 ; CHECK-NEXT:    srwi 3, 3, 28
-; CHECK-NEXT:    extsw 3, 3
+; CHECK-NEXT:    srwi 3, 3, 1
```

Comment:
If the `srwi` can not be merged with previous `srwi` with the changed mask, we need a follow up issue to merge these two RLWINM.

```
RLWINM 3, 3, 4, 28, 31
RLWINM 3, 3, 31, 31, 31
```

===>

```
RLWINM 3, 3, 3, 31, 31
```

Reply:
I created issue https://github.com/llvm/llvm-project/issues/104888 for this combine opportunity.